trs_filer/ga4gh/trs/endpoints/registerObjects.py

```diff
@@ -0,0 +1,73 @@
+"""Controller for adding new objects."""
+
+from random import choice
+from typing import Dict
+from trs_filer.app import logger
+from flask import (current_app, request)
+from pymongo.errors import DuplicateKeyError
+
+
+class CreateToolPostObject:
+    def __init__(self, request: request):
+        self.db_collection = (
+            current_app.config['FOCA'].db.dbs['trsStore'].
+            collections['objects'].client
+        )
+        self.tool_data = request.json
+        self.id_charset = current_app.config['FOCA'].\
+            endpoints['objects']['id_charset']
+        self.id_length = current_app.config['FOCA'].\
+            endpoints['objects']['id_length']
+        self.host_name = current_app.config['FOCA'].server.host
+
+    def create_id(self, charset, length) -> str:
+        """Creates random ID."""
+        return ''.join(choice(charset) for __ in range(length))
+
+    def create_tool_class(self):
+        """Create tool class."""
+        self.tool_data['toolclass'] = {
+            "description": "Temporary tool class.",
+            "id": "123456",
+            "name": "ToolClass"
+        }
+
+    def create_object(self) -> Dict:
+        """
+        Add new tool post objects to TRS Registry.
+
+        Args:
+            request: API request object.
+
+        Returns:
+            A unique identifier for the object.
+        """
+
+        # set checker variable
+        if "checker_url" in self.tool_data:
+            self.tool_data['has_checker'] = True
+        else:
+            self.tool_data['has_checker'] = False
+
+        # set tool class (proper method needs to be added)
+        self.create_tool_class()
+
+        # set meta_version method
+
+        # set unique id and url and save object
+        while True:
+            try:
+                generated_object_id = self.\
```

Comment:
Please don't wrap your lines like that, it's hard to read and error-prone. Especially here, there is absolutely no need to do it, because you can simply put your method params on different lines. Also, you con merge this line with the next (because you use the `generated_object_id` only once, when assigning it to something else). Finally, for added readability/maintainability, it's always good to be explicit, so if a function/method has named arguments, it's nice to _always_ call them by name, too. Then, if somebody rearranges the args in `.create_id()`, the call will still work.

This is better:

```py
                self.tool_data['id'] = self.create_id(
                    charset=self.id_charset,
                    length=self.id_length,
                )
```